lib/ClangImporter/ImportMacro.cpp

```diff
@@ -424,118 +440,158 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
+        secondValue = intValue->first;
+        secondType  = intValue->second;
+      }
+    }
+    // Couldn't parse an integer for the second value.
+    if (!secondType) { return nullptr; }
+
+    // Addition:
+    // (INT1 + INT2)
+    if (tokenI[1].is(clang::tok::plus)) {
+      llvm::APSInt value = firstValue + secondValue;
+      return createMacroConstant(impl, macro, name, DC, firstType,
                                  clang::APValue(value),
-                                 ConstantConvertKind::Coerce, /*isStatic=*/false,
-                                 ClangN);
-    // Check for an expression of the form (FLAG1 | FLAG2), (FLAG1 & FLAG2),
-    // (FLAG1 || FLAG2), or (FLAG1 || FLAG2)
-    } else if (tokenI[0].is(clang::tok::identifier) &&
-               isBinaryOperator(tokenI[1]) &&
-               tokenI[2].is(clang::tok::identifier)) {
-      auto firstID = tokenI[0].getIdentifierInfo();
-      auto secondID = tokenI[2].getIdentifierInfo();
-
-      if (firstID->hasMacroDefinition() && secondID->hasMacroDefinition()) {
-        auto firstMacroInfo = impl.getClangPreprocessor().getMacroInfo(firstID);
-        auto secondMacroInfo = impl.getClangPreprocessor().getMacroInfo(
-                                                                      secondID);
-        auto firstIdentifier =
-            impl.getNameImporter().importMacroName(firstID, firstMacroInfo);
-        auto secondIdentifier =
-            impl.getNameImporter().importMacroName(secondID, secondMacroInfo);
-        impl.importMacro(firstIdentifier, firstMacroInfo);
-        impl.importMacro(secondIdentifier, secondMacroInfo);
-        auto firstIterator = impl.ImportedMacroConstants.find(firstMacroInfo);
-        if (firstIterator == impl.ImportedMacroConstants.end()) {
-          return nullptr;
-        }
-        auto secondIterator = impl.ImportedMacroConstants.find(secondMacroInfo);
-        if (secondIterator == impl.ImportedMacroConstants.end()) {
-          return nullptr;
-        }
-
-        auto firstConstant = firstIterator->second;
-        auto secondConstant = secondIterator->second;
-        auto firstValue = firstConstant.first;
-        auto secondValue = secondConstant.first;
-        if (!firstValue.isInt() || !secondValue.isInt()) {
-          return nullptr;
+                                 ConstantConvertKind::Coerce,
+                                 /*isStatic=*/false, ClangN);
+    }
+    // Subtraction:
+    // (INT1 - INT2)
+    else if (tokenI[1].is(clang::tok::minus)) {
+      llvm::APSInt value = firstValue - secondValue;
+      return createMacroConstant(impl, macro, name, DC, firstType,
+                                 clang::APValue(value),
+                                 ConstantConvertKind::Coerce,
+                                 /*isStatic=*/false, ClangN);
+    }
+    // Left-shift:
+    // (INT1 << INT2)
+    else if (tokenI[1].is(clang::tok::lessless)) {
```

Comment:
If we're adding all these other things, does it make sense to support `>>` as well? I'm sure it's rare but even so.